bvm/ballerina-config/src/main/java/org/ballerinalang/config/ConfigProcessor.java

```diff
@@ -76,6 +76,39 @@ public static BConfig processConfiguration(Map<String, String> runtimeParams,
         return configEntries;
     }
 
+    private static void flatTomlMap(String currentPath, Object obj, Map<String, Object> map) {
+
+        if (obj instanceof String) {
+            String strObj = (String) obj;
+            map.put(currentPath, strObj);
+            return;
+        } else if (obj instanceof Long) {
+            Long longObj = (Long) obj;
+            map.put(currentPath, longObj);
+            return;
+        } else if (obj instanceof Double) {
+            Double doubleObj = (Double) obj;
+            map.put(currentPath, doubleObj);
+            return;
+        } else if (obj instanceof List) {
+            List listObj = (List) obj;
+            map.put(currentPath, listObj);
+            return;
+        } else if (obj instanceof Boolean) {
+            Boolean booleanObj = (Boolean) obj;
+            map.put(currentPath, booleanObj);
+            return;
+        }
+
+        if (obj instanceof Map) {
```

Comment:
Why is this if block separated from the other if-else block?

Reply:
Changed to single if-else block.